pkg/cloud/models/models.go

```diff
@@ -215,24 +216,55 @@ func (cp *CustomPricing) GetSharedOverheadCostPerMonth() float64 {
 	return sharedCostPerMonth
 }
 
-func SetCustomPricingField(obj *CustomPricing, name string, value string) error {
+func sanitizeFloatString(number string, allowNaN bool) (string, error) {
+	num, err := strconv.ParseFloat(number, 64)
+	if err != nil {
+		return "", fmt.Errorf("expected a string representing a number; got '%s'", number)
+	}
+	if !allowNaN && math.IsNaN(num) {
+		return "", fmt.Errorf("expected a string representing a number; got 'NaN'")
+	}
 
+	// Format the numerical string we just parsed.
+	// - Format as a floating point number, as we received it.
+	// - Precision 12 because storage can be expressed that precisely.
+	// - Use a bit size of 64.
+	return strconv.FormatFloat(num, 'f', 12, 64), nil
```

Comment:
Is storage the field requiring highest-precision today? Would we ever want this to be even higher precision than 12? (Or, vice-versa, lower precision?)

The documentation for `strconv.FormatFloat` says `The special precision -1 uses the smallest number of digits necessary such that ParseFloat will return f exactly.`. Why not use that? (I'm learning here, too).

Reply:
`-1` is a fantastic find. I'll use that. (I picked `12` to accord with the smallest number of the bunch, which was Storage.)